app/controllers/application_controller/buttons.rb

```diff
@@ -250,6 +251,17 @@ def applies_to_class_model(applies_to_class)
     applies_to_class == "ServiceTemplate" ? Service : applies_to_class.constantize
   end
 
+  def custom_button_done
+    url = SystemConsole.where(:vm => params[:id]).first.try(:url)
```

Comment:
Change to: `url = SystemConsole.find_by(:vm => params[:id]).try(:url)`

You are currently setting the `url` when calling `set_remote_console_url` from automate.  Do you need to store the `url` somewhere else since it already resides in the SystemConsole instance associated to the VM?

If so automate supports `custom_get` and `custom_set` methods.  [lib/miq_automation_engine/service_models/miq_ae_service_vm_or_template.rb#L140](https://github.com/ManageIQ/manageiq/blob/b31af139725839414a942607fe87a253624bdf91/lib/miq_automation_engine/service_models/miq_ae_service_vm_or_template.rb#L140)

Note: These custom attributes are visible in the VM summary screen.  Here is an example of what it would look like to set a custom attribute named `url`:
![image](https://cloud.githubusercontent.com/assets/2591569/19708794/379afab6-9aef-11e6-96b9-d02ea996b82a.png)